libs/libdaemon/src/BaseDaemon.cpp

```diff
@@ -804,7 +806,8 @@ void BaseDaemon::buildLoggers(Poco::Util::AbstractConfiguration & config)
         error_log_file->setProperty(Poco::FileChannel::PROP_PURGECOUNT, config.getRawString("logger.count", "10"));
         error_log_file->setProperty(Poco::FileChannel::PROP_FLUSH, config.getRawString("logger.flush", "true"));
         error_log_file->setProperty(Poco::FileChannel::PROP_ROTATEONOPEN, config.getRawString("logger.rotateOnOpen", "false"));
-        errorlog->setChannel(error_log_file);
+        error_log_file_async = new Poco::AsyncChannel(error_log_file);
```

Comment:
`AsyncChannel` will create a thread. It means that each time the `tiflash.toml` is modified, a thread will be created.
The previously created thread in `AsyncChannel` will still exist because the old `log` is not released.